Public/Src/Engine/Scheduler/Scheduler.cs

```diff
@@ -5354,6 +5364,14 @@ void IFileContentManagerHost.ReportMaterializedArtifact(in FileOrDirectoryArtifa
             m_pipOutputMaterializationTracker.ReportMaterializedArtifact(artifact);
         }
 
+        private void MakeSharedOpaqueOutputIfNeeded(AbsolutePath path, bool force = false)
+        {
+            if (force || PipGraph.IsPathUnderOutputDirectory(path))
+            {
+                SharedOpaqueOutputHelper.EnforceFileIsSharedOpaqueOutput(path.ToString(Context.PathTable));

```

Comment:
>EnforceFileIsSharedOpaqueOutput [](start = 41, length = 31)

Maybe now a better name would be 'EnforceFileIsUnderSharedOpaqueOutput'

Reply:
indeed